### PR TITLE
Adds the Bobr 12 Gauge Revolver to the Goodies category in cargo

### PR DESCRIPTION
**Pull request by Boviro**
## About The Pull Request
Note: I am working at the request of Aeri on discord, the "Good for the game" category is her reasoning.
Adds the Bobr 12 gauge revolver to cargo under the Goodies category, it costs 2000 credits, and comes with a box of shotgun shells to load it with.
I believe 2000 credits (2x the price of the Sol pistols) is a fair price for it, but I encourage others to weigh in on that value.

## Why It's Good For The Game
Because it's enjoyable to use and not overpowered, so there's not much reason not to.

## Proof Of Testing
No major changes were made, just adding a new supply pack. It has been tested locally and worked as intended.

## Changelog

:cl:
add: Adds the Bobr shotgun revolver to cargo as a goody.
/:cl:

### CODE REVIEW

**Comment by Boviro** [comment]
Note: I failed to create a new fork for this PR, but that fork is already identical to Bubberstation/Bubberstation aside from this PR's subject, so it has no consequence other than being a bit of clutter.

**Comment by Swiftfeather** [comment]
COMITTING FROM MASTER!!!!!!!!!!!!!!!!!!!!!! THE WHEEL SHALL DECIDE YOUR FATE

**Comment by Boviro** [comment]
oh nay

**Comment by Swiftfeather** [comment]
image

**Comment by NanoCats** [comment]
> image

actually I have the original verdict from earlier so now you GOTTA \s
image

**Comment by NanoCats** [comment]
Also I would mention that it's basically as far as I know, not a fantastic gun, it feels like it fires in a 180 degree range in front of your character.

File: modular_zubbers/code/modules/cargo/packs/goodies.dm [hunk 1]
```diff
@@ -37,6 +37,14 @@
 	access_view = ACCESS_WEAPONS
 	contains = list(/obj/item/gun/energy/e_gun/mini)
 
+/datum/supply_pack/goody/shotgun_revolver
+	name = "Bóbr 12 GA Revolver Single-Pack"
+	desc = "Contains 1 Bóbr revolver, chambered in 12 gauge. For when you really want the power of a shotgun in the palm of your hand. Comes with a box of beanbag shells."
+	contains = list(/obj/item/gun/ballistic/revolver/shotgun_revolver = 1,
+	/obj/item/ammo_box/advanced/s12gauge/buckshot = 1)
+	access_view = ACCESS_WEAPONS
+	cost = PAYCHECK_COMMAND * 15
```
Comment (projectkepler-ru) [inline modular_zubbers/code/modules/cargo/packs/goodies.dm:46]:
```suggestion
	cost = PAYCHECK_COMMAND * 18
```
This is literally cheaper than the actual shotguns you can get, worse or not, it shouldn't be, as it fit in your bag and still hold more shot

File: modular_zubbers/code/modules/cargo/packs/goodies.dm [hunk 2]
```diff
@@ -37,6 +37,14 @@
 	access_view = ACCESS_WEAPONS
 	contains = list(/obj/item/gun/energy/e_gun/mini)
 
+/datum/supply_pack/goody/shotgun_revolver
+	name = "Bóbr 12 GA Revolver Single-Pack"
+	desc = "Contains 1 Bóbr revolver, chambered in 12 gauge. For when you really want the power of a shotgun in the palm of your hand. Comes with a box of beanbag shells."
+	contains = list(/obj/item/gun/ballistic/revolver/shotgun_revolver = 1,
+	/obj/item/ammo_box/advanced/s12gauge/buckshot = 1)
```
Comment (projectkepler-ru) [inline modular_zubbers/code/modules/cargo/packs/goodies.dm:44]: This is not beanbag

**Review by projectkepler-ru — CHANGES_REQUESTED** [review]

File: modular_zubbers/code/modules/cargo/packs/goodies.dm (hunk 1 above)
Comment (NanoCats) [reply modular_zubbers/code/modules/cargo/packs/goodies.dm:46]: I was also going to suggest it be more money lol

**Comment by Boviro** [comment]
Yes, I did have a minor brain fart there, I had to rewrite the PR at least once and accidentally got the description and ammo wrong, but it's fixed now, and the price is 1800 credits (command paycheck * 18)

File: modular_zubbers/code/modules/cargo/packs/goodies.dm (hunk 1 above)
Comment (Boviro) [reply modular_zubbers/code/modules/cargo/packs/goodies.dm]: I wasn't certain how much it should be and the person who wanted it added didn't really specify, changed to 18x command paycheck now (1800 credits at this time)

File: modular_zubbers/code/modules/cargo/packs/goodies.dm (hunk 2 above)
Comment (Boviro) [reply modular_zubbers/code/modules/cargo/packs/goodies.dm]: This is on me, minor typo, fixed in the newest commit

**Comment by YakumoChen** [comment]
Shotgun revolver sounds pretty strong to me...

The HoS gets a special bulky (instead of huge) combat shotgun that has... 5 rounds? How is this not quite close in power level?

**Comment by YakumoChen** [comment]
Keep in mind this is a shotgun that is not two handed, does not need pumping, and fits in a backpack, downsides are four shot cylinder and sawn-off spread.

Honestly I think a lot of players would take this over a combat shotty.

**Comment by YakumoChen** [comment]
Not even to mention shotguns have always been considered one of the strongest weapons in the game the crew can easily get.

**Comment by Boviro** [comment]
Do you think it would be more balanced if it was spawned with beanbags in the chamber and with a 50% damage modifier? I'd like to get this gun in, specifically, with as little changes to its regular operation as I can. I do understand where you're coming from, randomly getting hit with a shotgun slug from a gun that fits in a bag isn't a very fun player experience.

**Comment by projectkepler-ru** [comment]
> Do you think it would be more balanced if it was spawned with beanbags in the chamber and with a 50% damage modifier? I'd like to get this gun in, specifically, with as little changes to its regular operation as I can. I do understand where you're coming from, randomly getting hit with a shotgun slug from a gun that fits in a bag isn't a very fun player experience.

Sure.

**Comment by Boviro** [comment]
Above commit adds a new shotgun_revolver subtype, shotgun_revolver\civvie , which has a 50% damage modifier, and spawns with beanbag rounds in the chamber. Replaces the Bobr pistol in the titular goody case with the new civvie variant. Also  increases the price further.

File: modular_zubbers/code/modules/cargo/packs/goodies.dm [hunk 3]
```diff
@@ -37,6 +37,14 @@
 	access_view = ACCESS_WEAPONS
 	contains = list(/obj/item/gun/energy/e_gun/mini)
 
+/datum/supply_pack/goody/shotgun_revolver
+	name = "Bóbr 12 GA Revolver Single-Pack"
+	desc = "Contains 1 civilian-modified Bóbr revolver, chambered in 12 gauge. For when you really want the power of a shotgun in the palm of your hand. Comes with a box of beanbag shells."
+	contains = list(/obj/item/gun/ballistic/revolver/shotgun_revolver/civvie = 1,
+	/obj/item/ammo_box/advanced/s12gauge/bean = 1)
+	access_view = ACCESS_WEAPONS
+	cost = PAYCHECK_COMMAND * 26
```
Comment (projectkepler-ru) [inline modular_zubbers/code/modules/cargo/packs/goodies.dm:46]:
```suggestion
	cost = PAYCHECK_COMMAND * 20
```
It shouldn't cost more than the m1911, they should be roughly equal in capability, especially if it's gonna have halved damage.

**Review by projectkepler-ru — APPROVED** [review]
Eeeh, beyond some how getting a distinct icon, this is okay

**Comment by NanoCats** [comment]
A 50% damage modifier is pretty steep, will it still be good for anything? I'm not a game balance scientist but like, it'd be lame if I bothered Boviro to make this thing so hard only for it not to actually be worth using. I was going to propose 0.90x 0.80x or 0.75x modifier. 

it _does_ have the enormous spread of a sawed off keep in mind.


One of my balance ideas was that it could damage the hand you fire it with a bit because it would provide a sensible drawback (realistic too, go shoot a Taurus Judge, I dare you)

**Comment by NanoCats** [comment]
> Eeeh, beyond some how getting a distinct icon, this is okay

Are you saying you want an alternate skin for it?

**Comment by Boviro** [comment]
I _could_ make a slightly modified sprite to signify it's the civvie variant (it's really not that hard), and I did think a 50% penalty is pretty steep, but I think the most difficult thing would be adding recoil damage, since AFAIK no other gun currently has it.

**Comment by NanoCats** [comment]
> Eeeh, beyond some how getting a distinct icon, this is okay

Would you accept 0.75x damage scalar, 0.50 seems like it would be a bit too steep. 

Keep in mind that we could always TM it first and then adjust it down further if needed.

**Comment by projectkepler-ru** [comment]
> > Eeeh, beyond some how getting a distinct icon, this is okay
> 
> Are you saying you want an alternate skin for it?

the standard revolver shotgun is a thing that you neeed to somehow know is different to this

If not, the change to the shotgun should just be made modular rather than a seperate weapon type if they won't be distinct

**Comment by projectkepler-ru** [comment]
> > Eeeh, beyond some how getting a distinct icon, this is okay
> 
> Would you accept 0.75x damage scalar, 0.50 seems like it would be a bit too steep.
> 
> Keep in mind that we could always TM it first and then adjust it further if needed.
> 
> I've consulted my balance wizard and they said it would make it literally worse than any other gun in the game
> 
> Also the HOS' shotgun holds 6, and it doesn't have the recoil, inaccuracy, and it fires quicker.
> 
> Also-also can't overstate how painful the inaccuracy/recoil is in an actual fight.

Testing this right now locally

File: tgstation.dme [hunk 4]
```diff
@@ -8861,6 +8861,9 @@
 #include "modular_zubbers\modules\modular_weapons\code\company_and_or_faction_based\carwo.dm"
 #include "modular_zubbers\modules\modular_weapons\code\company_and_or_faction_based\carwo_defense_systems\ammo\ammobox.dm"
 #include "modular_zubbers\modules\modular_weapons\code\company_and_or_faction_based\carwo_defense_systems\ammo\carwo.dm"
+
```
Comment (BurgerLUA) [inline tgstation.dme:8864]: there appears to be an empty line here

Comment (Boviro) [reply tgstation.dme:8864]: Indeed there does, but I'm not sure how it got there, as my local copy doesn't have it.

**Pull request merged**